front/lib/api/assistant/generation.ts

```diff
@@ -505,11 +504,14 @@ export async function constructPrompt(
     instructions += `\n${fallbackPrompt}`;
   }
 
-  const actionConfig = deprecatedGetFirstActionConfiguration(configuration);
-
-  if (isRetrievalConfiguration(actionConfig)) {
+  // If one of the action is a retrieval action, we add the retrieval meta prompt.
+  const hasRetrievalAction = configuration.actions.some((action) =>
+    isRetrievalConfiguration(action)
```

Comment:
I think we can iterate later, but I am wondering if this thing should really always be included if the agent has a retrieval action. Or maybe tweak it a bit, eg "if content  has been retrieved, focus on the retrieved content...."

Might be confusing for the model to say "focus on the retrieved content" if no retrieval has been executed.

@PopDaph @spolu

Reply:
Can we fork the implementation for MultiActions as this is something on which we'll want to iterate a lot (and we want to do it without risk of breaking the existing behavior).

Once we converge we'll be able to merge back

Reply:
I added a commit to fork.